scapy/plist.py

```diff
@@ -30,6 +30,8 @@
 
 class PacketList(BasePacketList, _CanvasDumpExtended):
```

Comment:
Could you explain why you are making this a (fake) socket ?

Reply:
I stumbled over the following problem:
During my unit test, I wanted to convert `CAN` frames from a `pcap` file into `ISOTPMessages`. 
`sniff` with a `ISOTPSession` works great here. On a second step, I wanted to feed this isotp messages to a `ECUSession`. 
To summarize, my use case could be solved, if it would be possible to concatenate `sniff` Sessions. 
I didn't saw an easy way to implement that. Also I didn't want to create a couple of specific sessions like (CAN->ISOTP->ECUSession or ISOTP->ECUSession). 
The next easiest solution for my problem, since my first `sniff` with an `ISOTPSession` already returns a `PacketList`, was the modification of `PacketList`, to be able to feed it to `sniff` with a different Session, again.

Reply:
I get your point.

A way of concenating sessions is something we'll probably add some day (to cleanup TCPsession that uses IPsession).

I don't like the way you have currently done it. We're starting to have slightly too many fake sockets.

I'll think of something else. If you have an idea, please let me know

Reply:
Do you have ideas how I can get forward with this approach to stack sessions together?

Reply:
I cleaned up `DefaultSession.__init__`. It's not so ugly anymore.  What do your think about this `supersession` parameter?